app/services/daily_xml_to_db_v2.rb

```diff
@@ -0,0 +1,205 @@
+require 'pp'
+# require 'sqlite3'
+require 'nokogiri'
+
+# CODE FLOW OUTLINE
+# -- PARSE XML FILES INTO AR OBJECTS --
+# get directory for xml files
+# get file list
+# loop while next file exists
+# open file with nokogiri
+# parse file
+# get all issues
+# construct db row for issues table
+# create AR record issue object from xml data and save it
+# get all pages in issues
+# construct db row for pages table
+# create AR record page object from xml data and save it
+# end loop
+# clean up file handling
+
+# IMPLEMENTATION
+
+class DailyXmlToDb_v2
+  attr_accessor :db, :db_name, :xml_dir, :file_list, :doc, :yml_file, :yml, :print_ok, :publication, :pub_id
+
+  def initialize(slug, testing)
+
+    if testing == "testing"
+      pp "TESTING..."
+      @print_ok = true
+    else
+      @print_ok = false
+    end
+
+    pp "Slug is: #{slug}"
+
+    # @pub_id = add_daily_publication
+    @pub_id = 1
+ 
+    # set directory for xml files
+    @xml_dir = "./tmp/xml_data/"
+
+    # get list of xml files
+    @xml_list = []
+    @xml_list = get_flist
+
+    pp_ok "File list:"
+    pp_ok @xml_list
+
```

Comment:
So then I would use, for example:
        text_link_b = text_link_a.xpath("@xlink:href", @NSMAP)
Correct?